src/compiler/scanner.ts

```diff
@@ -1828,6 +1829,9 @@ namespace ts {
                         pos++;
                         return token = SyntaxKind.GreaterThanToken;
                     case CharacterCodes.question:
+                        if (text.charCodeAt(pos + 1) === CharacterCodes.dot && !isDigit(text.charCodeAt(pos + 2))) {

```

Comment:
Nit: If you re-order the `pos++` to come before the `if` it marginally simplifies the `if`.